tests/cds-videos/conftest.py

```diff
@@ -9,34 +9,94 @@
+def create_app():
+    """Factory function for pytest-invenio."""
+
+    def factory(**config):
+        app = invenio_create_app(**config)
+        # Register custom blueprints like in cds-videos
+        app.register_blueprint(files_rest_blueprint)
+        app.register_blueprint(cds_api_blueprint)
+        return app
+
+    return factory
 
 
 @pytest.fixture(scope="module")
 def app_config(app_config):
     """Application configuration fixture."""
     base_path = os.path.dirname(os.path.realpath(__file__))
     logs_dir = os.path.join(base_path, "tmp/logs/")
-    if not os.path.exists(logs_dir):
-        os.makedirs(logs_dir)
-    app_config["MOUNTED_MEDIA_CEPH_PATH"] = os.path.join(
-        base_path, "data/files//media_data"
+    os.makedirs(logs_dir, exist_ok=True)
+
+    instance_path = tempfile.mkdtemp()
+    os.environ["INVENIO_INSTANCE_PATH"] = instance_path
+    os.environ["INVENIO_STATIC_FOLDER"] = os.path.join(
+        sys.prefix, "var/instance/static"
+    )
+
+    app_config.update(
+        {
+            "DEBUG_TB_ENABLED": False,
+            "TESTING": True,
+            "CELERY_TASK_ALWAYS_EAGER": True,
+            "CELERY_RESULT_BACKEND": "cache",
+            "CELERY_CACHE_BACKEND": "memory",
+            "CELERY_TASK_EAGER_PROPAGATES_EXCEPTIONS": True,
+            "CELERY_TASK_TRACK_STARTED": True,
+            "SQLALCHEMY_TRACK_MODIFICATIONS": False,
+            "SQLALCHEMY_DATABASE_URI": "postgresql+psycopg2://invenio:invenio@localhost/invenio",
+            "SITE_URL": "https://localhost:5000",
+            "JSONSCHEMAS_HOST": "cds.cern.ch",
+            "DEPOSIT_UI_ENDPOINT": "{scheme}://{host}/deposit/{pid_value}",
+            "PIDSTORE_DATACITE_DOI_PREFIX": "10.0000",
+            "ACCOUNTS_JWT_ENABLE": False,
+            "THEOPLAYER_LIBRARY_LOCATION": "https://localhost-theoplayer.com",
+            "THEOPLAYER_LICENSE": "CHANGE_ME",
+            "PRESERVE_CONTEXT_ON_EXCEPTION": False,
+            "REST_CSRF_ENABLED": False,
+            "MOUNTED_MEDIA_CEPH_PATH": os.path.join(base_path, "data/files/media_data"),
+            "CDS_MIGRATOR_KIT_LOGS_PATH": logs_dir,
+            "CDS_MIGRATOR_KIT_STREAM_CONFIG": "tests/cds-videos/data/streams.yaml",
+            "WEBLECTURES_MIGRATION_SYSTEM_USER": "weblecture-service@cern.ch",
+        }
     )
-    app_config["CDS_MIGRATOR_KIT_LOGS_PATH"] = logs_dir
-    app_config["CDS_MIGRATOR_KIT_STREAM_CONFIG"] = "tests/cds-videos/data/streams.yaml"
+
     return app_config
 
 
-@pytest.fixture(scope="module")
-def create_app(app_config):
-    """Create test app."""
-    return create_ui
+@pytest.fixture(scope="module", autouse=True)
+def db(app):
+    """Setup database."""
+    if not database_exists(str(db_.engine.url)):
+        create_database(str(db_.engine.url))
+    db_.create_all()
+    yield db_
+    db_.session.remove()
+    db_.drop_all()
+
+
+@pytest.fixture()
+def pidstore(app):
+    """Initialize invenio-indexer app."""
+    return InvenioPIDStore(app)
```

Comment:
comment: If this is required in most of the tests, it could be included directly in the app initialization.
```suggestion
@pytest.fixture()
def pidstore(app):
    """Initialize invenio-pidstore app."""
    return InvenioPIDStore(app)
```

Reply:
I was trying to do a similar conftest like in videos. `pidstore` is not used for now, but I thought we’ll need it for loading tests (checking legacy pid etc.). Now I checked the [videos tests](https://github.com/search?q=repo%3ACERNDocumentServer%2Fcds-videos%20pidstore%2C&type=code) again, and it’s not used there either. So I guess we don’t need it. WDYT?